app/services/open_library_service.rb

```diff
@@ -1,37 +1,31 @@
-require 'httparty'
-require 'json'
-
 class OpenLibraryService
-  include HTTParty
-  include Concurrent::Async
+  require 'httparty'
+  URL = Rails.application.credentials[:base_uri]
   def initialize(isbn)
     @isbn = isbn
-    @base_uri = Rails.application.credentials.development[:open_library][:base_uri]
   end
 
   def book_info
     response = find_book
-    if response.code == 200
-      hash = JSON.parse(response.body).with_indifferent_access
-      { response: build_response(hash), status: response.code }
-    else
-      { status: response.code, response: response.message }
-    end
+    return build_bad_response if response.parsed_response.blank?
+
+    hash = JSON.parse(response.body).with_indifferent_access
+    return { response: build_response(hash), status: response.code } if response.code == 200
```

Comment:
Since you delegated the bad response, you can also delegate to a different method the good response and keep this cleaner.